cis194/week1/tien/homework1/src/Lib.hs

```diff
@@ -0,0 +1,30 @@
+module Lib
+  ( toDigits
+  , toDigitsRev
+  , buildWeights
+  , doubleEveryOther
+  , sumDigits
+  , validate
+  ) where
+
+-- Exercise One: --
+toDigits :: Integer -> [Integer]
+toDigits num
+  | num > 0 = map (read . (: [])) $ show num :: [Integer]
+  | otherwise = []
+
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev num
+  | num > 0 = reverse $ toDigits num
+  | otherwise = []
+
+-- Exercise Two: --
+buildWeights :: Integer -> [Integer]
+buildWeights num = reverse $ map ((+ 1) . (`mod` 2)) [0 .. num - 1]
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther arr = zipWith (*) arr $(buildWeights . toInteger . length) arr
```

Comment:
Like the first comment this is considered better: `doubleEveryOther arr = zipWith (*) arr . buildWeights . toInteger . length  $ arr`.

Reply:
@shineli1984 can you please explain more on this comment? I don't quite get this `$ arr` I thought we use `$` to eval some expression but here it's only a variable `arr`. Thanks.